client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/TcpTransportFactory.java

```diff
@@ -92,6 +95,9 @@
    private Map<byte[], Boolean> compatibilityCaches = CollectionFactory
            .makeMap(ByteArrayEquivalence.INSTANCE, AnyEquivalence.getInstance());
 
+   private final BlockingQueue<AddClientListenerOperation> disconnectedListeners =
```

Comment:
How about using `LinkedBlockingQueue` here? It's based on linked list and we never know how many disconnected listeners we will have...